demos/higher_order_mass_lumping/higher_order_mass_lumping.py.rst

```diff
@@ -0,0 +1,183 @@
+
+In the work of :cite:`Kong:1999` and later :cite:`Geevers:2018`, several triangular and tetrahedral elements were discovered that could produce convergent and stable mass lumping for :math:`p \ge 2`. Here we implemented eight of them for use within the Firedrake computing environment (e.g., five triangular elements :math:`p \le 5` and three tetrahedrals up to :math:`p \le 3`). We honorifically refer to the elements as `KMV` elements (i.e., Kong-Mulder-Veldhuizen).
+
+We can then start our Python script loading Firedrake *and* fiat/finat. Fiat/finat are used to build the special quadrature rules that result in a diagonal mass matrix::
+
+    from firedrake import *
+    import FIAT
+    import finat
+
+    import math
+
+A simple triangular mesh is created::
+
+    mesh = UnitSquareMesh(50, 50)
+
+We choose a degree 2 `KMV` continuous function space and set it up, create a function to hold the seismic velocity, and then some functions used in time-stepping::
+
+    V = FunctionSpace(mesh, "KMV", 2)
+
+    u = TrialFunction(V)
+    v = TestFunction(V)
+
+    c = Function(V) # Scalar wave velocity
+
+    u_np1 = Function(V)  # timestep n+1
+    u_n = Function(V)    # timestep n
+    u_nm1 = Function(V)  # timestep n-1
+
+.. note::
+    The user can select orders up to P=5 for triangles and up to P=3 for tetrahedral!
+
+We create an output file to hold the simulation results::
+
+    outfile = File("out.pvd")
+
+Now we set the time-stepping variables performing a simulation for 1 second::
+
+    T = 1.0
+    dt = 0.001
+    t = 0
+    step = 0
+
+In seismology, often [Ricker wavelets](https://wiki.seg.org/wiki/Dictionary:Ricker_wavelet) are used to excite the domain, which have only one free parameter: a peak frequency :math:`freq`. Here we inject a Ricker wavelet into the domain with a frequency of 6 Hz. We also specify the seismic velocity in the domain :math:`c` to be a constant for simplicity::
+
+    freq = 6
+    c = Function(V).assign(1.5)
+
+The following two functions are used to inject the Ricker wavelet source into the domain::
+
+    # Source function
+    def RickerWavelet(t, freq, amp=1.0):
+        # Shift in time so the entire wavelet is injected
+        t = t - (math.sqrt(6.0) / (math.pi * freq))
+        return amp * (
+            1.0 - (1.0 / 2.0) * (2.0 * math.pi * freq) * (2.0 * math.pi * freq) * t * t
+        )
+
+
+    # Kernel function to apply the source function
+    def delta_expr(x0, x, y, sigma_x=2000.0):
+        sigma_x = Constant(sigma_x)
+        return exp(-sigma_x * ((x - x0[0]) ** 2 + (y - x0[1]) ** 2))
```

Comment:
I agree. This injects the wavelet into the mesh. 

In regard to the first arg: I did that in this way because the kwarg argument doesn't have to be declared a `Constant` by the user.

Reply:
I added a blurb describing this part of the code.